validation/helpers.py

```diff
@@ -0,0 +1,195 @@
+from os import fspath
+
+import divvunspell
+from difflib import Differ, SequenceMatcher, get_close_matches
+import operator
+
+import requests
+import json
+
+from django.conf import settings
+
+"""
+RULES
+
+    adding/removing diacritics or hyphens, swapping glides w/y, or 
+    adding/removing aspirations -h- between vowel and consonant would have a weight of zero.
+    inserting the vowel -i- between two consonants would have a half-weight (0.5).
+    inserting/removing/swapping any other characters would have a normal weight of one.
+
+"""
+
+
+archive = divvunspell.SpellerArchive(fspath(settings.BASE_DIR / "crk.zhfst"))
+speller = archive.speller()
+
+vowels = ["a", "i", "o", "â", "î", "ô"]
+consonants = ["  c", "  h", "  k", "  m", "  n", "  p", "  s", "  t", "  w", "  y"]
+
+
+def get_edit_distance(word):
+
+    ranked_suggestions = speller.suggest(word)
+    suggestions = [s[0] for s in ranked_suggestions]
+
+    rankings = {}
+    for s in suggestions:
+        med = get_differ(word, s)
+        rankings[s] = med
+
+    sorted_tuples = sorted(rankings.items(), key=operator.itemgetter(1))
+    sorted_rankings = {k: v for k, v in sorted_tuples}
+
+    return sorted_rankings
+
+
+def get_differ(word, suggestion):
+    med = 0
+    dif = Differ()
+    df = list(dif.compare(word, suggestion))
```

Comment:
Since `dif` is never used elsewhere, the code might be cleaner as

```suggestion
    df = list(Differ().compare(word, suggestion))
```

instead of having two variables right next to each other with very similar names